tasks/tests/unit/test_sync_repo_languages.py

```diff
@@ -0,0 +1,218 @@
+from datetime import datetime, timedelta
+from unittest.mock import Mock
+
+import pytest
+from shared.torngit.exceptions import TorngitError
+from shared.utils.enums import TaskConfigGroup
+
+from database.tests.factories.core import OwnerFactory, RepositoryFactory
+from tasks.sync_repo_languages import SyncRepoLanguagesTask
+
+MOCKED_NOW = datetime(2024, 7, 3, 6, 8, 12)
+LIST_WITH_INTERSECTION = ["python", "go", "javascript"]
+
+
+def setup_now(mocker):
+    mocker.patch(
+        f"tasks.{TaskConfigGroup.sync_repo_languages.value}.get_utc_now",
+        return_value=MOCKED_NOW,
+    )
+
+
+@pytest.fixture
```

Comment:
A bit new to fixtures as we don't use them in API as much, wdyt? Moreso for my own learning

Reply:
Idk about antipatterns....(everything after this is my own opinion and can be disregarded) but not everything needs to be an explicit fixture. That makes it harder to know what it is in very large test files. Usually we use decorators.

This `setup_with_languages` seems fine to me. It's re-used in many tests, and I feel that's what fixtures are used for. At the same time it's harder for me reading the test to know the inputs cause I'd have to see the definition of the fixture.

Usually I prefer to try and make the test as self-contained as possible, which means adding the mocks in the test function directly (again, only my opinion)

SoI'd have just used the `@patch` decorator for the   the `get_repo_provider_service` and added the return value in the test function directly.

Also for the time mock you can use https://pypi.org/project/pytest-freezegun/ apparently (it's in the requirements.txt at least)

Reply:
Ty for the input, I honestly like having tests more self-contained as well so I'm happy to keep tests that way moving forward. Would you like me to do that here?